magicctapipe/scripts/lst1_magic/lst1_magic_stereo_reco.py

```diff
@@ -54,39 +58,45 @@
 logger.setLevel(logging.INFO)
 
 
-def calculate_pointing_separation(event_data):
+def calculate_pointing_separation(event_data, config):
     """
-    Calculates the angular distance of the LST-1 and MAGIC pointing
+    Calculates the angular distance of the LST and MAGIC pointing
     directions.
 
     Parameters
     ----------
     event_data: pandas.core.frame.DataFrame
-        Data frame of LST-1 and MAGIC events
-
+        Data frame of LST and MAGIC events
+    config: dict
+        Configuration for the LST + MAGIC analysis
     Returns
     -------
     theta: pandas.core.series.Series
-        Angular distance of the LST-1 and MAGIC pointing directions
-        in the unit of degree
+        Angular distance of the LST array and MAGIC pointing directions
+        in units of degree
     """
-
-    # Extract LST-1 events
-    df_lst = event_data.query("tel_id == 1")
-
-    # Extract the MAGIC events seen by also LST-1
-    df_magic = event_data.query("tel_id == [2, 3]")
+    
+    assigned_tel_ids = config["mc_tel_ids"] #This variable becomes a dictionary, e.g.: {'LST-1': 1, 'LST-2': 0, 'LST-3': 0, 'LST-4': 0, 'MAGIC-I': 2, 'MAGIC-II': 3}
+    LSTs_IDs = np.asarray(list(assigned_tel_ids.values())[0:4])
```

Comment:
like I commented below - this is dangerous because it relies on the sequence of the telescope names in the config file

Reply:
See the previous reply about the check_input_list function.